code/datums/action.dm

```diff
@@ -137,8 +139,9 @@
 	if(viewers[our_hud]) // Already have a copy of us? go away
 		return
 
-	LAZYOR(viewer.actions, src) // Move this in
-	ShowTo(viewer)
+	if(owner_has_control)
+		LAZYOR(viewer.actions, src) // Move this in
+		ShowTo(viewer)
```

Comment:
This seems like the improper place to put this, if I understand it correctly - shouldn't be in `Grant` instead, before `GiveAction(M)`?

Reply:
I want them to have the action, just simply not be able to control it.

Reply:
They'll still BE the owner of the action, they'll just not be given the action button in `grant()`